src/authorities/QueueAuthority.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// This class is responsible for managing queue state based on instances running
+
+import RabbitMQChannelFactory from '../factories/RabbitMQChannelFactory';
+import {getLogger} from '../logger';
+import config from '../config';
+import PS2AlertsInstanceInterface from '../interfaces/PS2AlertsInstanceInterface';
+import {PS2EventInstanceHandlerContract} from '../interfaces/PS2EventInstanceHandlerContract';
+import {TYPES} from '../constants/types';
+import {injectable, multiInject} from 'inversify';
+import ZoneMessageHandler from '../handlers/ps2census/ZoneMessageHandler';
+import {ChannelWrapper} from 'amqp-connection-manager';
+import InstanceAbstract from '../instances/InstanceAbstract';
+
+@injectable()
+export default class QueueAuthority {
+    private static readonly logger = getLogger('QueueAuthority');
+    private readonly instanceChannelMap = new Map<InstanceAbstract['instanceId'], ChannelWrapper[]>();
+    private readonly handlerMap = new Map<string, Array<PS2EventInstanceHandlerContract<any>>>();
+
+    constructor(
+        private readonly channelFactory: RabbitMQChannelFactory,
+        @multiInject(TYPES.eventInstanceHandlers) eventInstanceHandlers: Array<PS2EventInstanceHandlerContract<any>>,
+    ) {
+        for (const handler of eventInstanceHandlers) {
+            let handlerList = this.handlerMap.get(handler.eventName);
+
+            if (!handlerList) {
+                handlerList = [];
+                this.handlerMap.set(handler.eventName, handlerList);
+            }
+
+            handlerList.push(handler);
+        }
+    }
+
+    public startQueuesForInstance(instance: PS2AlertsInstanceInterface): void {
+        // Check if we're already monitoring the instance, if we are, do nothing
+        if (this.instanceChannelMap.has(instance.instanceId)) {
+            return;
+        }
+
+        const channels: ChannelWrapper[] = [];
+
+        for (const [eventName, handlers] of this.handlerMap) {
+            const channel = this.channelFactory.create(
+                config.rabbitmq.topicExchange,
+                `aggregator-${instance.world}-${eventName}`,
```

Comment:
Does it? We're only subscribing to the world, this isn't being seperated out by Instance ID as we're doing instance ID checking upon receiving the message

Reply:
That only applies for MetagameEvent where one singular event is required to be processed and resolved by a singular aggregator. If you have 2+ aggregators consuming a world queue, it shouldn't matter which consumer picks it up as it'll at the end of the day resolve to an instance based off the world and zone combo.

The challenge may lay in how to perform instance management between 2+ aggregators. Currently the list of instances are kept in memory, instead this may need to be pushed into a shared redis DB where whichever consumer gets the MetagameEvent is responsible for updating.

Then, at the point we check the instance via `getInstance(world, zone)` that simply just changes to be a redis call to get instances by world and zone.

Reply:
However, just for now, we are still maintaining a 1:1 relationship with the Census environment, the work to make it multiple aggregators will come at another time, it's too complex to put in this already monstreous PR.